### PR TITLE
feat(test): test fault-recovery on madsim

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR adds fault-recovery to the deterministic e2e test.

When running `./risedev sslt` with `--kill-node`, the test runner will randomly kill a compute node between queries.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#2646, #4527

### CODE REVIEW

**Review by BowenXiao1999 — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

**Review by BugenZhao — APPROVED** [review]

File: ci/scripts/deterministic-e2e-test.sh [hunk 1]
```diff
@@ -25,3 +25,10 @@ timeout 90s cargo make sslt --release -- -j 16 './e2e_test/streaming/**/*.slt'
 
 echo "--- deterministic simulation e2e, ci-3cn-2fe, parallel, batch"
 timeout 30s cargo make sslt --release -- -j 16 './e2e_test/batch/**/*.slt'
+
+# bugs here!
```
Comment (BugenZhao) [inline ci/scripts/deterministic-e2e-test.sh:29]: 👀👀👀

**Comment by wangrunji0408** [comment]
TPCH Q19 failed: https://buildkite.com/singularity-data/pull-request/builds/6736#0182cf11-3740-4cc4-b3a1-bb69a186aa9b/216-489

Is it an uncommon bug?

@mergifyio requeue

**Comment by wangrunji0408** [comment]
@mergifyio requeue

File: ci/scripts/deterministic-e2e-test.sh (hunk 1 above)
Comment (BugenZhao) [reply ci/scripts/deterministic-e2e-test.sh:29]: Do we have an issue for this?

Comment (wangrunji0408) [reply ci/scripts/deterministic-e2e-test.sh:29]: Not yet. I'm confirming it and will open one once I find the cause.

**Comment by wangrunji0408** [comment]
@mergifyio requeue

**Comment by BugenZhao** [comment]
```
Failed to build executor: internal error: unsupported aggregator: type=SingleValue input=Timestamp output=Timestamp distinct=false
```

Looks like the fuzzing tests are using an unsupported aggregator. Is this expected? cc @kwannoel

**Comment by kwannoel** [comment]
> ```
> Failed to build executor: internal error: unsupported aggregator: type=SingleValue input=Timestamp output=Timestamp distinct=false
> ```
> 
> Looks like the fuzzing tests are using an unsupported aggregator. Is this expected? cc @kwannoel 

If it's unsupported it should be disabled, will take a look later. Thanks for catching this.

**Pull request merged**